src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -929,28 +916,46 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
     }
 
     @api
-    reset(objectMappingDeveloperName = null, setDefaults = true) {
-        const sectionsList = this.template.querySelectorAll('c-ge-form-section');
-
-        let fieldMappingDevNames = null;
+    reset(objectMappingDeveloperName = null) {
         if (objectMappingDeveloperName === null) {
-            this.dataImport = null;
-            this.setReviewDonationsDonorProperties(null);
-            this.resetStoredDonationDonorProperties();
+            this.resetFieldsApplyDefaults();
+            this.resetLocalState();
         } else {
-            fieldMappingDevNames =
-                Object.values(GeFormService.fieldMappings).filter(
-                    ({Target_Object_Mapping_Dev_Name, DeveloperName}) =>
-                        Target_Object_Mapping_Dev_Name === objectMappingDeveloperName)
-                    .map(({DeveloperName}) => DeveloperName);
+            this.resetFieldsForObjMappingApplyDefaults(objectMappingDeveloperName);
+            this.setFormStateToInitialFieldValuesForObjMapping(objectMappingDeveloperName);
         }
-
-        sectionsList.forEach(section => {
-            section.reset(fieldMappingDevNames, setDefaults);
-        });
         this.widgetData = {};
     }
 
+    resetLocalState() {
```

Comment:
What do you think about calling this function `resetFormState` ?

Reply:
Yeah, I like this suggestion also.  I initially intentionally did not call it that because it is (currently) not just resetting/initializing the new formState object, it is also updating state that is currently stored in these other local variables (and I just extracted most of this logic from its previous location to reduce the size of the function.  But, as we continue with this work, the calls to these methods on 931, 932, and 933 should go away since "reset" will only need to call this.initializeState once all state is stored in the formState object.  So, I'll update it, thanks!